files/class-vip-filesystem-stream-wrapper.php

```diff
@@ -694,9 +694,25 @@ public function mkdir( $path, $mode, $options ) {
 	 * @return  bool
 	 */
 	public function stream_metadata( $path, $option, $value ) {
-		$this->debug( sprintf( 'stream_metadata =>  %s + %s + %s', $path, $option, $value ) );
+		$this->debug( sprintf( 'stream_metadata =>  %s + %s + %s', $path, $option, json_encode( $value ) ) );   // phpcs:ignore WordPress.WP.AlternativeFunctions.json_encode_json_encode
 
-		return false;
+		switch ( $option ) {
+			case STREAM_META_TOUCH:
+				if ( false === file_exists( $path ) ) {
+					$file = fopen( $path, 'w' );    // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_fopen
+					if ( is_resource( $file ) ) {
+						fclose( $file );
+						return true;
+					}
+
+					return false;
+				}
```

Comment:
This seems to be the simplest way to implement `touch()`: `file_exists()` will skip downloading the file if it exists; `fopen()` will create an empty file, and then `fclose()` will make sure that the empty file is uploaded to the server.

If we skip `file_exists()`, I suspect the system will download the file if it exists, which might not be desirable for large files.